samples/Quickstart/src/Program.cs

```diff
@@ -38,24 +42,30 @@
             services.UseTelemetry(config.InstrumentationKey);
         }
 
-        // Used for accessing Azure resources
-        services.UseAuthenticator();
-
         // Setup custom headers for use in an Input Filter
         services.UseCustomHeaders();
         services.AddCustomHeader("X-MS-AZUREFHIR-AUDIT-USER-TOKEN-TEST", "QuickstartCustomOperation", CustomHeaderType.RequestStatic);
 
+        
+        services.AddAzureClients(clientBuilder =>
+        {
+            clientBuilder.AddGenericRestClient(new Uri(config.FhirServerUrl))
+            .WithCredential(new DefaultAzureCredential(true))
+            .ConfigureOptions(options =>
+            {
+                options.Retry.Mode = Azure.Core.RetryMode.Fixed;
+                options.Retry.MaxRetries = 3;
+                options.Retry.MaxDelay = TimeSpan.FromSeconds(120);
+            });
+        });
         // Setup pipeline for Azure function
         services.UseAzureFunctionPipeline();
 
         // Add our header modification as the first filter
         services.AddInputFilter(typeof(QuickstartFilter));
 
         // Add our binding to pass the call to the FHIR service
-        services.AddBinding<RestBindingOptions>(typeof(RestBinding), options =>
-        {
-            options.ServerUrl = config.FhirServerUrl;
-        });
+        services.AddRestBinding();
```

Comment:
Since GenericRestClient is called by RestBinding, it would be nice if the configuration for these was joined - AddRestBinding method would take in the Uri, the credential, and the client options.